server/src/test/java/org/elasticsearch/action/admin/indices/create/TransportCreatePartitionsActionTest.java

```diff
@@ -60,116 +54,94 @@ public void prepare() {
-    }
+        execute("create table table1 (a int, b int) " +
+            "partitioned by (a) " +
+            "clustered into 1 shards " +
+            "with (number_of_replicas=0)");
 
-    @Test
-    public void testCreateBulkIndicesIgnoreExistingSame() throws Exception {
-        PutIndexTemplateRequest templateRequest = new PutIndexTemplateRequest("*")
-            .patterns(List.of("*"))
-            .mapping(Map.of())
-            .settings(Settings.builder()
-                .put("number_of_shards", 1)
-                .put("number_of_replicas", 0)
-            );
-        cluster().client().execute(PutIndexTemplateAction.INSTANCE, templateRequest).get();
-        List<String> indices = Arrays.asList("index1", "index2", "index3", "index1");
-        AcknowledgedResponse response = action.execute(
-            new CreatePartitionsRequest(indices)
-        ).get();
-        assertThat(response.isAcknowledged(), is(true));
+        ensureYellow();
         Metadata indexMetadata = cluster().clusterService().state().metadata();
-        for (String index : indices) {
-            assertThat(indexMetadata.hasIndex(index), is(true));
+
+        // CREATE TABLE... PARTITIONED BY doesn't create an index, it creates only template via MetadataIndexTemplateService.
+        assertThat(indexMetadata.indices()).isEmpty();
+
+        // Create some indices/partitions
+        execute("insert into table1 (a,b) values (1,1), (2,2), (3,3)");
+        execute("refresh table table1");
+        Metadata updatedMetadata = cluster().clusterService().state().metadata();
+        List<String> indexUUIDs = new ArrayList<>();
+        for (ObjectCursor<String> cursor: updatedMetadata.indices().keys()) {
+            indexUUIDs.add(updatedMetadata.index(cursor.value).getIndexUUID());
         }
-        AcknowledgedResponse response2 = action.execute(
-            new CreatePartitionsRequest(indices)
-        ).get();
-        assertThat(response2.isAcknowledged(), is(true));
+        assertThat(indexUUIDs.size()).isEqualTo(3); // 1 table with 3 partitions.
+
+        // Try to insert into same partitions, when index is created it should not be re-created
+        // Only 1 new index should be created here, for a partition 4.
+        execute("insert into table1 (a,b) values (1,1), (2,2), (3,3), (4,4)");
+        execute("refresh table table1");
+
+        ClusterState currentState = cluster().clusterService().state();
+        updatedMetadata = currentState.metadata();
+        List<String> newUUIDs = new ArrayList<>();
+        for (ObjectCursor<String> cursor: updatedMetadata.indices().keys()) {
+            newUUIDs.add(updatedMetadata.index(cursor.value).getIndexUUID());
+        }
+        assertThat(newUUIDs.size()).isEqualTo(4); // 1 table with 4 partitions.
+        assertThat(newUUIDs).containsAll(indexUUIDs); // old indices are still there, they were not overwritten
```

Comment:
It's ok for me to merge these 2 tests but then the new tests should:
 - have a valid name expressing what is tested, e.g. `test_insert_into_existing_partition_does_not_recreate_it`
 - validate what it was meant to do: it wasn't about testing that it stays with 4 partitions, but to test that the partition wasn't recreated, such an existing row should not be lost.

Reply:
updated in https://github.com/crate/crate/pull/13807/commits/75d6c8ecab19ee94b08485d505a8e9ad3d3f0e0b

it's a single test, `testCreateBulkIndicesIgnoreExistingSame` is removed as it's basically same with `testRoutingOfIndicesIsNotOverridden`